packages/backend/src/core/discovery/runDiscovery.ts

```diff
@@ -1,20 +1,22 @@
 import { MainnetEtherscanClient } from '@l2beat/common'
 import { providers } from 'ethers'
 
-import { DiscoveryConfig } from '../../config/Config'
+import { DiscoveryModuleConfig } from '../../config/Config'
 import { ConfigReader } from './ConfigReader'
 import { discover } from './discover'
 import { DiscoveryLogger } from './DiscoveryLogger'
 import { ProviderWithCache } from './provider/ProviderWithCache'
 import { saveDiscoveryResult } from './saveDiscoveryResult'
+import { getDiscoveryConfigHash } from './utils/getDiscoveryConfigHash'
 
 export async function runDiscovery(
   provider: providers.AlchemyProvider,
   etherscanClient: MainnetEtherscanClient,
   configReader: ConfigReader,
-  config: DiscoveryConfig,
+  config: DiscoveryModuleConfig,
 ) {
   const projectConfig = await configReader.readConfig(config.project)
```

Comment:
I am thinking about extracting configReader to common package as I am also reading discovery in deriving